### ISSUE TITLE
Navigator does not update packages

### ISSUE THREAD

**Issue by yazabaza** [ContinuumIO/anaconda-issues#12445]
Issue 9087 was marked as Closed but still exists. It is 2 years old and users are still reporting the same problem.

https://github.com/ContinuumIO/anaconda-issues/issues/9087

**Comment by robert99** [comment]
Navigator marks a series of packages as updateable.
The update runs but does not update the package.
What is not clear is whether that means there is a new package but your particular system cannot update to it because of older supporting packages. Anaconda does not give any warning that this may be the case. So a user is expecting it to update but it never does.
I have also had cases when a package which would update has downloaded and updated other packages that I did not expect.
Very difficult to know what is going on.
Sometimes conda (from Navigator) can update packages, sometime it cannot.

**Comment by BabakAmini** [comment]
I came here from that closed issue and have the same issue. I tried all the solutions I found on the internet but I am still unable to update some of the packages.

**Comment by yazabaza** [comment]
I just submitted the same issue again. Third time in 2 years. I think it's a lost cause. I gave up on Navigator long ago and just conda on command line. Anaconda Navigator is a POS.

**Comment by robert99** [comment]
I did a complete clean uninstall (delete all files related to conda and spyder etc) and reinstalled the November version.
I then used conda to update everything.
Navigator is now on version 4 and everything seems to work now.

There is also a recent new Anaconda version you might try.

**Comment by macsunmood** [comment]
What if updating via conda through terminal doesn't help as well?
Should I exhaustively switch to pip for this case?

**Comment by robert99** [comment]
pip can work or it might clash with conda versions.
Anaconda is so complex now that the chances of things not being compatible gets higher and higher.
Although Navigator keeps telling you there are upgrades - when you try to install them they just effectively say the latest one that will work with your set up is already installed.
I think users will have to accept the version they start with as Anaconda is not reliably updateable.
It may be that you should ignore the install advice and use the set paths option.
It may be in Windows that there is a space in your user name - this fault was cured but perhaps they forgot that.
It may be that you have to switch off your antivirus software during installs.

Last week there was a new Anaconda version.
I suggest you clean uninstall and try that - most things are up to date.
Otherwise you will be spending a lot of time getting nowhere. 

Here are some of my notes on clean uninstalling on my system:

![image](https://user-images.githubusercontent.com/1282053/143442552-93882f83-1ea9-4061-906e-32b195ca7dc8.png)

**Comment by macsunmood** [comment]
@robert99 Thanks for your detailed answer. My version of Anaconda was also the latest. Uninstalled it and reinstalled it again pure fresh. Can't update/install almost nothing among packages. What does one do when conda does this for hours with zero outcome?
![image](https://user-images.githubusercontent.com/45285672/143579072-1463e778-0ee1-46d6-ae8f-532a591a345b.png)

**Comment by robert99** [comment]
Yes, your problems are very similar to what I have been getting for the last two years.
Frozen solve and flexible solve find dozens of already installed packages that are not compatible with your intended upgrade so it fails. I find it hard to believe that so many unrelated packages could be incompatible.
Anaconda do not seem interested in solving it, nor even explaining what is going on.
If you are using pip to install a program that has not been installed in the Anaconda package then this still works for me.
Otherwise, users have to accept that Anaconda is no longer updateable.
Just install it, and if it works, leave it alone.

What you could try is WinPython
https://winpython.github.io/
It has a large range of the latest packages and you can use pip to update anything.
Any missing programs, such as Tensorflow or PyTorch, you need can often be found ready compiled at 
https://www.lfd.uci.edu/~gohlke/pythonlibs/
These are also easy to install.
WinPython also has the advantage that it is run as is and unbreakable - i.e. not installed to Windows - so is transportable and runs every time from the installed directory.
The community is very helpful.

**Comment by macsunmood** [comment]
@robert99 thanks again! Would you recommend WinPython over Anaconda specifically for DS/ML purpose? Looks very interesting, but at the same time lack of official support from Tensorflow / PyTorch and such makes it dubious.

**Comment by robert99** [comment]
There are about 24,000 posts on TensorFlow in their Community - that is some support!
https://github.com/issues?q=tensorflow

Designed for scientists, data-scientists, and education (thanks to NumPy, SciPy, Sympy, Matplotlib, Pandas, pyqtgraph, etc.):
interactive data processing and visualization using Python with Spyder and Jupyter/IPython, Pyzo, IDLEX or IDLE
fully integrated Cython and Numba! See included example
connectors (cffi, odbc, rpy2, scilab2py, requests, ...) for advanced users.
So it is ideal for DS/ML

Anaconda Navigator has an excellent interface and I use it when it works.
When it fails I use WinPython. 

There are alternatives to try:
ActivePython, "CPython", Enthought Canopy (new system now). 
Active Python has Tensorflow etc
https://www.activestate.com/products/python/python-machine-learning/
